### PR TITLE
allow update of additionalTags in AWSFargateProfile

### PR DESCRIPTION
**Pull request by abhinavnagaraj**
**What type of PR is this?**

**What this PR does / why we need it**:
Allows update of additionalTags in AWSFargateProfile

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/cluster-api-provider-aws/pull/26#" title="Author self-approved">abhinavnagaraj</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign  after the PR has been reviewed.
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Fcluster-api-provider-aws).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/cluster-api-provider-aws/blob/spectro-master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: exp/api/v1alpha3/awsfargateprofile_webhook.go [hunk 1]
```diff
@@ -93,6 +88,35 @@ func (r *AWSFargateProfile) ValidateUpdate(oldObj runtime.Object) error {
...
+				allErrs,
+				field.Invalid(field.NewPath("spec", name), update, "field is immutable"),
+			)
+		}
+	}
+
+	appendErrorIfMutated(old.Spec.ClusterName, r.Spec.ClusterName, "clusterName")
+	appendErrorIfMutated(old.Spec.ProfileName, r.Spec.ProfileName, "profileName")
+	appendErrorIfMutated(old.Spec.SubnetIDs, r.Spec.SubnetIDs, "subnetIDs")
+	appendErrorIfSetAndMutated(old.Spec.RoleName, r.Spec.RoleName, "roleName")
```
Comment (jzhoucliqr) [inline exp/api/v1alpha3/awsfargateprofile_webhook.go:114]: I think RoleName is optional?

File: exp/api/v1alpha3/awsfargateprofile_webhook.go [hunk 2]
```diff
@@ -75,12 +75,7 @@ func (r *AWSFargateProfile) ValidateUpdate(oldObj runtime.Object) error {
 	}
 
 	var allErrs field.ErrorList
-	if !reflect.DeepEqual(old.Spec, r.Spec) {
-		allErrs = append(
-			allErrs,
-			field.Invalid(field.NewPath("spec"), r.Spec, "is immutable"),
-		)
-	}
+	allErrs = append(allErrs, r.validateImmutable(old)...)
```
Comment (jzhoucliqr) [inline exp/api/v1alpha3/awsfargateprofile_webhook.go:78]: these change together with https://github.com/spectrocloud/cluster-api-provider-aws/pull/12/files.  can you please create a issue in upstream, and cherry-pick for an upstream master PR?

Comment (abhinavnagaraj) [reply exp/api/v1alpha3/awsfargateprofile_webhook.go:78]: create a PR/commit in spectrocloud combining these 2 and then open an upstream PR with that ?

File: exp/api/v1alpha3/awsfargateprofile_webhook.go (hunk 1 above)
Comment (abhinavnagaraj) [reply exp/api/v1alpha3/awsfargateprofile_webhook.go:114]: not sure, this is how its handled in awsmanagedmachinepool

File: exp/api/v1alpha3/awsfargateprofile_webhook.go (hunk 2 above)
Comment (jzhoucliqr) [reply exp/api/v1alpha3/awsfargateprofile_webhook.go:78]: yes please. We need these when we upgrade to v1alpha4.

File: exp/api/v1alpha3/awsfargateprofile_webhook.go (hunk 1 above)
Comment (jzhoucliqr) [reply exp/api/v1alpha3/awsfargateprofile_webhook.go:114]: ok. I think it is for the first time controller to set the default roleName when spec.RoleName is empty.

**Review by jzhoucliqr — APPROVED** [review]
/approve

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/cluster-api-provider-aws/pull/26#" title="Author self-approved">abhinavnagaraj</a>*, *<a href="https://github.com/spectrocloud/cluster-api-provider-aws/pull/26#pullrequestreview-729830393" title="Approved">jzhoucliqr</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign  after the PR has been reviewed.
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Fcluster-api-provider-aws).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/cluster-api-provider-aws/blob/spectro-master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**